### PR TITLE
add combine config for metrics in pivot table

### PR DESCRIPTION
**Pull request by roganw**
Add combine config for metrics in pivot table.
If checkbox of 'combine_metric' is chose, Metrics will be combined together below the columns.

### CODE REVIEW

File: superset/assets/javascripts/explore/stores/controls.jsx [hunk 1]
```diff
@@ -240,6 +240,13 @@ export const controls = {
     description: 'Sort bars by x labels.',
   },
 
+  combine_metric: {
+    type: 'CheckboxControl',
+    label: 'Combine Metrics',
+    default: false,
+    description: 'Combine Metrics to display.',
```
Comment (mistercrunch) [inline superset/assets/javascripts/explore/stores/controls.jsx]: It's unclear to me what this does and why it's useful

Comment (RoganW) [inline superset/assets/javascripts/explore/stores/controls.jsx]: It's useful to compare data for different scenarios, when there exist multi metrics and they are relevant.

original behavior:


after select 'Combine Metrics'

Comment (mistercrunch) [inline superset/assets/javascripts/explore/stores/controls.jsx]: Oh I understand, we need a clearer description here. `Display metrics side by side within each column, as opposed to each column being displayed side by side for each metric.`

File: superset/viz.py [hunk 2]
```diff
@@ -384,6 +384,25 @@ def query_obj(self):
         d['groupby'] = list(set(groupby) | set(columns))
         return d
 
+    def combine_metric(self, df):
```
Comment (mistercrunch) [inline superset/viz.py]: I think you can do all this in 2 lines of code using stack/unstack. https://pandas.pydata.org/pandas-docs/stable/reshaping.html#reshaping-by-stacking-and-unstacking

**Review by mistercrunch — CHANGES_REQUESTED** [review]
Sorry about the delayed getting this properly reviewed.

File: superset/viz.py (hunk 2 above)
Comment (RoganW) [inline superset/viz.py]: Thank you, I will check that later.

Comment (RoganW) [inline superset/viz.py]: Yeah, you are right.
Modified done.

File: superset/viz.py [hunk 3]
```diff
@@ -402,6 +402,9 @@ def get_data(self, df):
             aggfunc=self.form_data.get('pandas_aggfunc'),
             margins=self.form_data.get('pivot_margins'),
         )
+        # Display metrics side by side with each column
+        if self.form_data.get('combine_metric'):
+            df = df.stack(0).unstack()
```
Comment (mistercrunch) [inline superset/viz.py]: Ah! that looks much better!

**Comment by mistercrunch** [comment]
I'll be happy to merge this once you correct the minor linting error in the javascript

**Comment by RoganW** [comment]
@mistercrunch Corrected

**Pull request merged**